### ISSUE TITLE
Rule proposal: `no-array-with`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2039]
### Description

[Array#with](https://github.com/tc39/proposal-change-array-by-copy#overview)

I don't know who thought it's add good idea to check the index range. (Checked history, but only one issue questioned it, and closed by owner)

```js
            if (actualIndex < 0 || actualIndex >= len) {
                throw new RangeError();
            }
```
https://github.com/tc39/proposal-change-array-by-copy/blob/e8771393a522dd7ed8ea16f31e92a29b22843afc/polyfill.js#L264

I agree it's safer, but this never happen in other array methods. (except the size of array is too big)

I havn't try to use it, just an idea.

### Fail

[TBD]

### Pass

[TBD]
